src/harness/BenchmarkDotNet.Extensions/ExclusionFilter.cs

```diff
@@ -27,4 +29,41 @@ public bool Predicate(BenchmarkCase benchmarkCase)
             return !globFilter.Predicate(benchmarkCase);
         }
     }
+
+    class CategoryExclusionFilter : IFilter
+    {
+        private readonly List<(string userValue, Regex regex)> patterns;
+
+        public CategoryExclusionFilter(List<string> patterns)
+        {
+            if (patterns != null)
+            {
+                this.patterns = patterns.Select(pattern => (pattern, new Regex(WildcardToRegex(pattern), RegexOptions.IgnoreCase | RegexOptions.CultureInvariant))).ToList();
+            }
+            else
+            {
+                patterns = null;
+            }
+        }
+
+        public bool Predicate(BenchmarkCase benchmarkCase)
+        {
+            if(patterns == null)
+            {
+                return true;
+            }
+            foreach (var category in benchmarkCase.Descriptor.Categories)
+            {
+                if(patterns.Any(pattern => category.Equals(pattern.userValue, StringComparison.OrdinalIgnoreCase) || pattern.regex.IsMatch(category)))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // https://stackoverflow.com/a/6907849/5852046 not perfect but should work for all we need
+        private static string WildcardToRegex(string pattern) => $"^{Regex.Escape(pattern).Replace(@"\*", ".*").Replace(@"\?", ".")}$";
```

Comment:
similarly to `ExclusionFilter` from #1307 you should be able to reuse the existing  [AnyCategoriesFilter](https://github.com/dotnet/BenchmarkDotNet/blob/master/src/BenchmarkDotNet/Filters/AnyCategoriesFilter.cs) from BenchmarkDotNet and just negate the output:

```suggestion
        private readonly AnyCategoriesFilter categoriesFilter;

        public CategoryExclusionFilter(List<string> patterns) => categoriesFilter = new AnyCategoriesFilter(patterns.ToArray());

        public bool Predicate(BenchmarkCase benchmarkCase) => !categoriesFilter.Predicate(benchmarkCase);
```

Reply:
Sounds good.